src/createPersistoid.js

```diff
@@ -37,7 +37,7 @@ export default function createPersistoid(config: PersistConfig): Persistoid {
     //if any key is missing in the new state which was present in the lastState,
     //add it for processing too
     Object.keys(lastState).forEach(key => {
-      if (state[key] === undefined) {
+      if (lastState[key] !== undefined && state[key] === undefined) {
```

Comment:
this check makes sense. Can you double check how this will interact with the recently merged: https://github.com/rt2zz/redux-persist/pull/928/files

Your tests look good, and run and pass. So my guess is we can simply add `lastState[key] !== undefined` as one more item in the conditional.